chain/chain/src/apply_chunks_thread_pool.rs

```diff
@@ -23,7 +23,7 @@ type IdleThreadQueue = Arc<Mutex<VecDeque<oneshot::Sender<Option<Job>>>>>;
 /// configured `priority`. Realtime threads **always** take precedence over
 /// threads using normal policy (`SCHED_OTHER`), so `priority` applies **only
 /// among other realtime threads**.
-pub(crate) struct ThreadPool {
+pub struct ThreadPool {
```

Comment:
People were really uncomfortable with what this TP does exactly and so it isn't clear to me if they would be comfortable with making it available generally.

Having proposed it in the first place, I personally think it would not be too terrible to make this generally usable. But at the same time it being extremely narrowly purposed allows modifying it for a given use-case without worrying when other uses might break. After this PR that particular benefit is lost.

Lets try to make an executive decision as to whether we are now comfortable with having a thread pool like this broadly available and if there are still reservations, consider duplicating this code for the new use-cases.